Rebus.AdoNet/AdoNetSagaPersister.cs

```diff
@@ -235,780 +140,127 @@ public AdoNetSagaPersisterFluentConfigurer CustomizeSerializationTypeResolving(F
-						var indexValuesParm = dialect.EscapeParameter(SAGAINDEX_VALUES_COLUMN);
-						var forUpdate = GetSagaLockingClause(dialect);
-						var valuesPredicate = ArraysEnabledFor(dialect)
-							? dialect.FormatArrayAny($"i.{indexValuesCol}", indexValuesParm)
-							: $"(i.{indexValuesCol} LIKE ('%' || {indexValuesParm} || '%'))";
-
-						command.CommandText = $@"
-							SELECT s.{dataCol}
-							FROM {sagaTblName} s
-							JOIN {indexTblName} i on s.{sagaIdCol} = i.{indexIdCol}
-							WHERE s.{sagaTypeCol} = {sagaTypeParam}
-								AND i.{indexKeyCol} = {indexKeyParam}
-								AND (
-									CASE WHEN {indexValueParm} IS NULL THEN i.{indexValueCol} IS NULL
-									ELSE 
-										(
-											i.{indexValueCol} = {indexValueParm}
-												OR
-											(i.{indexValuesCol} is NOT NULL AND {valuesPredicate})
-										)
-									END
-								  )
-							{forUpdate};".Replace("\t", "");
-
-						var value = GetIndexValue(fieldFromMessage);
-						var values = value == null ? null : ArraysEnabledFor(dialect)
-							? (object)(new[] { value })
-							: GetConcatenatedIndexValues(new[] { value });
-						var valuesDbType = ArraysEnabledFor(dialect) ? DbType.Object : DbType.String;
-
-						command.AddParameter(indexKeyParam, sagaDataPropertyPath);
-						command.AddParameter(sagaTypeParam, sagaType);
-						command.AddParameter(indexValueParm, DbType.String, value);
-						command.AddParameter(indexValuesParm, valuesDbType, values);
-					}
-
-					string data = null;
-
-					try
-					{
-						data = (string)command.ExecuteScalar();
-					}
-					catch (DbException ex)
-					{
-						// When in no-wait saga-locking mode, inspect
-						// exception and rethrow ex as SagaLockedException.
-						if (useSagaLocking && useNoWaitSagaLocking)
-						{
-							if (dialect.IsSelectForNoWaitLockingException(ex))
-								throw new AdoNetSagaLockedException(ex);
-						}
-
-						throw;
-					}
-
-					if (data == null) return null;
-
-					try
-					{
-						return JsonConvert.DeserializeObject<TSagaData>(data, Settings);
-					}
-					catch { }
-
-					try
-					{
-						return (TSagaData)JsonConvert.DeserializeObject(data, Settings);
-					}
-					catch (Exception exception)
-					{
-						var message = string.Format("An error occurred while attempting to deserialize '{0}' into a {1}", data, typeof(TSagaData));
-
-						throw new ApplicationException(message, exception);
-					}
-				}
-			}
+			return JsonConvert.SerializeObject(sagaData, Formatting.Indented, Settings);
 		}
 
-		private bool ArraysEnabledFor(SqlDialect dialect)
+		public virtual void Insert(ISagaData sagaData, string[] sagaDataPropertyPathsToIndex)
 		{
-			return useSqlArrays && dialect.SupportsArrayTypes;
+			throw new NotImplementedException("Insert method implementation missing?1");
 		}
 
-		private bool ShouldIndexValue(object value)
+		public virtual void Update(ISagaData sagaData, string[] sagaDataPropertyPathsToIndex)
```

Comment:
We should probably make this ones into an abstract.

Reply:
oki